edb/pgsql/metaschema.py

```diff
@@ -541,7 +553,7 @@ class StrToDecimal(dbops.Function):
                     NULL::numeric,
                     'invalid_text_representation',
                     msg => (
-                        'invalid syntax for numeric: '
+                        'invalid syntax for type numeric: '
```

Comment:
Does it make sense to have different implementations for bigint/decimal casts?

Reply:
They have different behavior

Reply:
I see that, but the mechanisms are different -- one checks for NaN and other for exceptions. Is that because of how we implement the bigint domain (i.e. it raises?)